cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -753,14 +794,36 @@ def handle_etcd_relation(reldata):
     api_opts.add('etcd-servers', connection_string, strict=True)
 
 
-def configure_master_services():
-    ''' Add remaining flags for the master services and configure snaps to use
-    them '''
-
+def get_config_args():
+    db = unitdata.kv()
+    old_config_args = db.get('api-extra-args', [])
+    # We have to convert them to tuples becuase we use sets
+    old_config_args = [tuple(i) for i in old_config_args]
+    new_config_args = []
+    new_config_arg_names = []
+    for arg in hookenv.config().get('api-extra-args', '').split():
+        new_config_arg_names.append(arg.split('=', 1)[0])
+        if len(arg.split('=', 1)) == 1:  # handle flags ie. --profiling
```

Comment:
https://kubernetes.io/docs/admin/kube-apiserver/ suggests flags without values might be supported for example:
--contention-profiling                                    Enable lock contention profiling, if profiling is enabled

Other arguments document expected value for example:
--etcd-cafile string
--audit-log-maxsize int